src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -2305,6 +2308,16 @@ NetworkOPsImp::recvValidation(
     std::shared_ptr<STValidation> const& val,
     std::string const& source)
 {
+    // Don't relay validations from another network.
+    if (auto networkID = app_.overlay().networkID())
+    {
+        auto const view = m_ledgerMaster.getCurrentLedger();
+
+        if (view->rules().enabled(featureNetworkIDValidation) &&
+            val->getNetworkID() != static_cast<Json::UInt>(*networkID))
```

Comment:
This code is probably wrong. The cast is unnecessary and confusing but (if my understanding is correct) the call `val->getNetworkID()` which calls `getFieldU32` which calls `STObject::getFieldByValue` will `throw` an exception if the `sfNetworkID` field is not present in the received validation.

Reply:
This is my first attempted commit to rippled and still trying to pick up C, so also learning lots as I go here. That being said will, test and validate this and possibly also add a test case to prove that we get the desired result here.